test/jdk/sun/net/www/http/KeepAliveCache/B8291637.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8291637
+ * @run main/othervm -Dhttp.keepAlive.time.server=20 -esa -ea B8291637
```

Comment:
Is it intentional that we are setting the -esa and -ea options here?

Reply:
If we enable assertions without the bug fix, then an assertion is triggered in KeepAliveCache, which is useful.

Reply:
tests are usually run with `-esa -ea` in our test systems, but individual developers could "forget" to specify those options. So in the case where it's the only thing that would make the test fail without the fix (or in the case where it would provide additional diagnostic for the particular bug fix) it can make sense to add them in the `@run` too...